iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -135,9 +135,25 @@ Viewer.prototype._bindHandlers = function () {
     $('#iframe-container .zoom-in').click(function () { viewer.zoomIn() });
     $('#iframe-container .zoom-out').click(function () { viewer.zoomOut() });
 
+    // Listen to messages posted to this window
+    $(window).on("message", function(e) { viewer.handleMessage(e) });
+
     TableExport.addHandles(this._contents, this._report);
 }
 
+Viewer.prototype.handleMessage = function (event) {
+    var jsonString = event.originalEvent.data;
+    var data = JSON.parse(jsonString);
+
+    if(data.task == 'SHOW_FACT') {
```

Comment:
Nit: repo not fully consistent, but space in `if (` seems 10x more prevalent.

Reply:
nit addressed